libblkid/src/topology/ioctl.c

```diff
@@ -48,18 +54,20 @@ static int probe_ioctl_tp(blkid_probe pr,
 		const struct topology_val *val = &topology_vals[i];
 		int rc = 1;
 		union {
-			unsigned long ul;
 			int i;
 			uint64_t u64;
 		} data;
 
 		if (ioctl(pr->fd, val->ioc, &data) == -1)
 			goto nothing;
 
+		if (val->kernel_size == 4)
+			data.u64 = data.i;
```

Comment:
Maybe add some comment to explain it ;-)

Reply:
Ack